common/p_setup.cpp

```diff
@@ -1462,8 +1464,51 @@ void P_LoadBlockMap (int lump)
 	blockmap = blockmaplump+4;
 }
 
-
-
+/*
+* @brief P_GenerateUniqueMapHash
+* 
+* Creates a unique map hash used to identify a unique map.
+* Based on a few key lumps that makes a map unique.
+* 
+* @param things - Things lump offset
+* @param linedefs - Linedefs lump offset
+* @param sidedefs - Sidedefs lump offset
+* @param vertexes - Vertexes lump offset
+* @param segs - Segs lump offset
+* @param ssectors - SSectors lump offset
+* @param sectors - Sectors lump offset
+*/
+void P_GenerateUniqueMapHash(int things, int linedefs, int sidedefs, int vertexes,
+                             int segs, int ssectors, int sectors)
+{
+	unsigned int length = 0;
+
+	typedef std::vector<byte> LevelLumps;
+	static LevelLumps levellumps;
+
+	const byte* thingbytes = static_cast<byte*>(W_CacheLumpNum(things, PU_STATIC));
+	const byte* lindefbytes = static_cast<byte*>(W_CacheLumpNum(linedefs, PU_STATIC));
+	const byte* sidedefbytes = static_cast<byte*>(W_CacheLumpNum(sidedefs, PU_STATIC));
+	const byte* vertexbytes = static_cast<byte*>(W_CacheLumpNum(vertexes, PU_STATIC));
+	const byte* segsbytes = static_cast<byte*>(W_CacheLumpNum(segs, PU_STATIC));
+	const byte* ssectorsbytes = static_cast<byte*>(W_CacheLumpNum(ssectors, PU_STATIC));
+	const byte* sectorsbytes = static_cast<byte*>(W_CacheLumpNum(sectors, PU_STATIC));
+
+	levellumps.insert(levellumps.end(), W_LumpLength(things), *thingbytes);
+	levellumps.insert(levellumps.end(), W_LumpLength(linedefs), *lindefbytes);
+	levellumps.insert(levellumps.end(), W_LumpLength(sidedefs), *sidedefbytes);
+	levellumps.insert(levellumps.end(), W_LumpLength(vertexes), *vertexbytes);
+	levellumps.insert(levellumps.end(), W_LumpLength(segs), *segsbytes);
+	levellumps.insert(levellumps.end(), W_LumpLength(ssectors), *ssectorsbytes);
+	levellumps.insert(levellumps.end(), W_LumpLength(sectors), *sectorsbytes);
+
+	length = W_LumpLength(things) + W_LumpLength(linedefs) + W_LumpLength(sidedefs) +
+	         W_LumpLength(vertexes) + W_LumpLength(segs) + W_LumpLength(ssectors) +
+	         W_LumpLength(sectors);
+
+	memcpy(::level.level_fingerprint, W_FARMHASH(levellumps.data(), length),
```

Comment:
This is a memory leak as-written.  `W_FARMHASH` is using malloc to allocate memory and returning a pointer to that memory.  You are then treating that return value as a temporary and discarding the pointer afterwards.

I believe that the correct fix here is to fix W_FARMHASH proper.  See my comments on that.

Reply:
Fixed this to be a bit more sane.